### ISSUE TITLE
Sitemap Generate Hanging

### ISSUE THREAD

**Issue by mortscode** [nystudio107/craft-seomatic#1236]
### Describe the bug
I have an entry type (`postEntries`) that has ~900 entries in it. When my sitemaps generate in the background, it seems to get stuck. When I navigate to the url for my sitemap everything works as expected. However, when I go to the sitemap for my post entries after it's generated in the background, the page hangs and eventually times out.

For this entry type, images are added to the sitemap.

- sitemap: [https://themodernproper.com/sitemaps.xml](https://themodernproper.com/sitemaps.xml)
- post entries sitemap: [https://themodernproper.com/sitemaps-1-section-postEntries-1-sitemap.xml](https://themodernproper.com/sitemaps-1-section-postEntries-1-sitemap.xml)

When I run the `./craft seomatic/sitemap/generate --handle=postEntries` script, everything is fixed and the sitemap is available after it generates.

### To reproduce
Steps to reproduce the behaviour:
1. After automatic generation, go to "[...](https://themodernproper.com/sitemaps-1-section-postEntries-1-sitemap.xml)"
2. If the page loads, it's because I ran the script manually, otherwise the page will hang and eventually timeout.

### Expected behaviour
Sitemaps generate automatically and successfully upon saved entry.

### Versions
- Plugin version: 3.4.41
- Craft version: 3.7.59

**Comment by khalwat** [comment]
So three things:

1) Probably you should ensure you are running your Craft queue via a queue runner as per: https://nystudio107.com/blog/robust-queue-job-handling-in-craft-cms Since you're using servd.host, you are, so this is mostly for other people reading this

2) When you hit the sitemap on the frontend before it has been regenerated via the queue job, it tries to generate the sitemap right then and there. However, if you have a ton of entries, it can time out (as you noted) based on your web server settings, just like any frontend request. 

3) You probably don't want all of your entries in the sitemap anyway. Sitemaps are used as a way to tell search engines how to discover new pages, they don't determine what is indexed or not. So if you remove something from your sitemap, it will still be in the search engine's index. So what many people do is just set the sitemap to have XX entries in it. SEOmatic will automatically sort by last modified, so the search engine will use its crawl budget to crawl the newly changed pages

TL;DR be patient and let the queue job regenerate the sitemap for you in the background, and consider only having the most recent 20 or so entries in your per-section sitemaps
